src/ArborX_DistributedSearchTree.hpp

```diff
@@ -12,6 +12,10 @@
 #ifndef ARBORX_DISTRIBUTED_SEARCH_TREE_HPP
 #define ARBORX_DISTRIBUTED_SEARCH_TREE_HPP
 
+#include <ArborX_Config.hpp>
+
+#ifdef ArborX_ENABLE_MPI
```

Comment:
Should we start explicitly listing our headers to avoid having to do such thing?